base/linalg/symmetric.jl

```diff
@@ -403,7 +403,54 @@ function svdvals!{T<:Real,S}(A::Union{Hermitian{T,S}, Symmetric{T,S}, Hermitian{
     return sort!(vals, rev = true)
 end
 
-#Matrix-valued functions
+#Matrix functions
+function ^{T<:Real}(A::Symmetric{T}, p::Integer)
+    if p < 0
+        return Symmetric(Base.power_by_squaring(inv(A), -p))
+    else
+        return Symmetric(Base.power_by_squaring(A, p))
+    end
+end
+function ^{T<:Real}(A::Symmetric{T}, p::Real)
+    F = eigfact(full(A))
```

Comment:
(In general, a lot of the `full` calls seem odd to me; we seem better off adding additional methods elsewhere, but I thought that could be left for another PR.  This particular one is especially egregious though because `eigfact` should be especially efficient for real-symmetric matrices.   What happens if you just delete all of the `full` calls?)